### ISSUE TITLE
Support for Intel oneAPI

### ISSUE THREAD

**Issue by mathomp4** [NVIDIA/hpc-container-maker#430]
This is both a question and maybe feature request. I wanted to make sure that the current intel_psxe building block doesn't work for Intel oneAPI, right? If not, I might try to make a PR to do it...but my Python is not good. :)

**Comment by samcmill** [comment]
Correct, the `intel_psxe` building block is for Intel Parallel Studio, before it was rebranded as OneAPI.

Intel now provides apt / yum repositories for OneAPI, so installation should be quite a bit more straight forward.

At one point, this was working (it may still be, I haven't retested it):

```
Stage0 += packages(ospackages=['ca-certificates', 'gnupg', 'wget'])

Stage0 += packages(
                   apt_keys=['https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB'],
                   apt_repositories=['deb https://apt.repos.intel.com/oneapi all main'],
                   ospackages=['intel-hpckit-2021.1.0'],
                   yum_keys=['https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2023.PUB'],
                   yum_repositories=['https://yum.repos.intel.com/oneapi'])
```

**Comment by mathomp4** [comment]
@samcmill Interesting. Okay. If I have time, maybe I'll try and make some `intel_oneapi` and `intel_oneapi_runtime` building blocks based on `packages`. I just know in the past it was so nice to have things done for me "behind the scenes" 😄
